File: app/src/main/java/com/spidroid/starry/adapters/PostInteractionListener.java
package com.spidroid.starry.adapters;

import android.view.View;
import com.spidroid.starry.models.PostModel;
import com.spidroid.starry.models.UserModel;
import java.util.List;

public interface PostInteractionListener {
    // التفاعلات الأساسية
    void onLikeClicked(PostModel post);
    void onCommentClicked(PostModel post);
    void onRepostClicked(PostModel post);
    void onBookmarkClicked(PostModel post);

    // قائمة الخيارات
    void onMenuClicked(PostModel post, View anchorView);

    // خيارات من القائمة
    void onTogglePinPostClicked(PostModel post);
    void onEditPost(PostModel post);
    void onDeletePost(PostModel post);
    void onCopyLink(PostModel post);
    void onSharePost(PostModel post);
    void onEditPostPrivacy(PostModel post);
    void onReportPost(PostModel post);

    // تفاعلات الريأكشنات
    void onLikeButtonLongClicked(PostModel post, View anchorView);
    void onEmojiSelected(PostModel post, String emojiUnicode);
    void onEmojiSummaryClicked(PostModel post);

    // تفاعلات أخرى
    void onHashtagClicked(String hashtag);
    void onPostLongClicked(PostModel post);
    void onMediaClicked(List<String> mediaUrls, int position);
    void onVideoPlayClicked(String videoUrl);
    void onLayoutClicked(PostModel post);
    void onSeeMoreClicked(PostModel post);
    void onTranslateClicked(PostModel post);
    void onShowOriginalClicked(PostModel post);

    // تفاعلات المستخدم
    void onUserClicked(UserModel user);
    void onFollowClicked(UserModel user);
    void onModeratePost(PostModel post);
}

File: app/src/main/java/com/spidroid/starry/adapters/PostInteractionListener.kt
package com.spidroid.starry.adapters

import android.view.View
import com.spidroid.starry.models.PostModel
import com.spidroid.starry.models.UserModel

interface PostInteractionListener {
    fun onLikeClicked(post: PostModel?)
    fun onCommentClicked(post: PostModel?)
    fun onRepostClicked(post: PostModel?)
    fun onQuoteRepostClicked(post: PostModel?)
    fun onBookmarkClicked(post: PostModel?)
    fun onMenuClicked(post: PostModel?, anchorView: View?)
    fun onTogglePinPostClicked(post: PostModel?)
    fun onEditPost(post: PostModel?)
    fun onDeletePost(post: PostModel?)
    fun onCopyLink(post: PostModel?)
    fun onSharePost(post: PostModel?)
    fun onEditPostPrivacy(post: PostModel?)
    fun onReportPost(post: PostModel?)
    fun onLikeButtonLongClicked(post: PostModel?, anchorView: View?)
    fun onReactionSelected(post: PostModel?, emojiUnicode: String)
    fun onEmojiSummaryClicked(post: PostModel?)
    fun onHashtagClicked(hashtag: String?)
    fun onPostLongClicked(post: PostModel?)
    fun onMediaClicked(mediaUrls: MutableList<String?>?, position: Int, sharedView: View)
    fun onVideoPlayClicked(videoUrl: String?)
    fun onLayoutClicked(post: PostModel?)
    fun onSeeMoreClicked(post: PostModel?)
    fun onTranslateClicked(post: PostModel?)
    fun onShowOriginalClicked(post: PostModel?)
    fun onModeratePost(post: PostModel?)
    fun onUserClicked(user: UserModel?)
    fun onFollowClicked(user: UserModel?)
}